crates/sparrow-runtime/src/execute/operation/scan.rs

```diff
@@ -156,6 +156,20 @@ impl ScanOperation {
                 ))?,
         );
 
+        // Figure out the projected columns from the table schema.
+        //
+        // TODO: Can we clean anything up by changing the table reader API
```

Comment:
I'm sure... but the scan needs some attention in a follow-up. Probably not needed *now*.